### PR TITLE
Compile book instructions

### LINKED ISSUE #23: Plonk Book / Implementation Documentation
Add an extensive documentation in the style of [Halo2 Book](https://zcash.github.io/halo2/index.html).
It should contain:

- Theoretical background and explanation of Plonk
- Details of the implementation including: description of the structures and traits, complexitites of the main operations, differences from the original protocol, etc.
- Examples of use

The purpose of the book should be to allow new developers to learn how to use this library in the easiest way possible.

### PR DESCRIPTION
**Pull request by mathcrypto**

### CODE REVIEW

**Comment by CPerezz** [comment]
@mathcrypto Can you rebase on top of `main` so that the history is not massive??
Otherways there are an insane amount of commits because you're re-using the same branch for different PR's.

File: plonk-book/src/SUMMARY.md [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 - [Chapter 1: Background Material](./chapter_1.md)
 - [Chapter 2: PLONK](./chapter_2.md)
-- [Chapter 3: PLONK library](./chapter_3.md)
+- [Chapter 3: ARK-PLONK library](./chapter_3.md)
```
Comment (CPerezz) [inline plonk-book/src/SUMMARY.md]: I'd leave as `PLONK` for now :)

File: plonk-book/src/chapter_3.md [hunk 2]
```diff
@@ -1,35 +1,33 @@
-# PLONK library
+# ARK-PLONK library
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 3]
```diff
@@ -1,35 +1,33 @@
...
 
 
-PLONK is a generic Rust PLONK implementation using arkworks as a backend. This PLONK library is one of many projects implementing PLONK like: 
-TurboPlonk, UltraPlonk, Plonky, ShPlonk, PLOOKUP, PLONKUP, halo2 etc.
+ARK-PLONK is a generic Rust PLONK implementation using arkworks as a backend. ARK-PLONK is one of many projects implementing PLONK like: TurboPlonk, UltraPlonk. DuskPlonk, Plonky, ShPlonk, PLOOKUP, PLONKUP, halo2...etc.
 
-PLONK is however the only generic implementation which allows any curve implementation or commitment scheme to be used and isn’t 
-restricted to only one implementation like other existing libraries.
+ARK-PLONK is however the only generic implementation which allows any curve implementation or commitment scheme to be used and isn’t restricted to only one implementation like other existing libraries (for example duskPlonk is based on bls12-281).
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 4]
```diff
@@ -1,35 +1,33 @@
...
 Halo based schemes have decent proof size and prover time, but take linear time to verify and it’s not possible to verify on Ethereum.
 
 Mir has recently developed a more optimized proving system Plonky2 based on PLONK and FRI. 
 FRI-based ZK-STARKs algorithm provides both quantum-resistance and does not require any trusted setups while the KZG scheme uses elliptic curve pairing which is not quantum resistant and requires a third-party trusted setup. FRI has a blowup factor which measures how much redundancy a polynomial needs to add before the commitment is generated and thus makes the prover faster. 
 Plonky2 claims a 100x speed up for ethereum for $170ms$ comparing to Plonky which takes $15s$ for proving times. 
-This PLONK repo is an optimization of the original PLONK protocol and guarantees $3.44s$ for proving time and $4.50ms$ for verifier speed comparing to $60$ proof times in the original PLONK design.
+ARK-PLONK is an optimization of the original PLONK protocol and guarantees $3.44s$ for proving time and $4.50ms$ for verifier speed comparing to $60$ proof times in the original PLONK design.
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 5]
```diff
@@ -38,7 +36,7 @@ Our implementation also uses custom gates similarly to [TurboPolnk](https://docs
 The original Plonk proposes simple circuits whose gates only perform arithmetic operations: additions and multiplications. Plonk is able to extend an arithmetic "gate" and allows the creation of new customized gates which can perform so many different operations (EC additions, logical XOR, efficient Poseidon or Pedersen hashes....etc).
 
 
-The different types of gates in this implementation are:
+The different types of gates in Ark-plonk are:
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 6]
```diff
@@ -1,35 +1,33 @@
...
+There is no other library right which allows you to have the freedom of using generic parameters. Zcash's Halo2 and Plonky use HALO commitment scheme while plonky2 uses only FRI scheme. In terms of elliptic curves, Aztek's implementation of PLonk is based on bn256, duskPlonk is based on bls12-281.
 ## Circuit implementation
 
-The implementation is an optimization of the original PLONK protocol as it enables lookup table to the PLONK circuit. This optimization allows for precomputation of some of the operations that are not snark friendly like bit operations (see [PLOOKUP](https://eprint.iacr.org/2020/315.pdf) for further explanation on PLONK + LOOKUP tables).
+ARK-PLONK's implementation is an optimization of the original PLONK protocol as it enables lookup table to the PLONK circuit. This optimization allows for precomputation of some of the operations that are not snark friendly like bit operations (see [PLOOKUP](https://eprint.iacr.org/2020/315.pdf) for further explanation on PLONK + LOOKUP tables).
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 7]
```diff
@@ -75,7 +73,7 @@ In order to use one of these gates, we set the value of its associated SP to $1$
 #### Design custom gates
 Before we explain how each of the previous mentioned custom gates are designed, we need to talk about fan-in 3 gates and why we use them at ark-plonk instead of fan-in 2 gates?
 The original Plonk design uses fan-in 2 gates which means each gates has two inputs and an output wires.
-These PLONK gates, on the other hand, are fan-in-3 gates, so they have one more wire which makes it 4 wires in total.
+Ark-plonk gates on the other hand are fan-in-3 gates, so they have one more wire which makes it 4 wires in total.
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 8]
```diff
@@ -107,7 +105,7 @@ PI
 = 0.
 $$
 
-This repository still allows the use of fan-in 2 gates only by setting the 4th wire selector, $q_d$, to $0$. 
+Ark-plonk still allows the use of fan-in 2 gates only by setting the 4th wire selector $q_d$ to $0$.
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 9]
```diff
@@ -158,7 +156,7 @@ We now have 21 quads and $21 \div 4 = 5$ remainder 1, so we will need 5 full gat
        
 ### Logic gates
 
-The plonk logic gates can perform bitwise AND and XOR operations. 
+Ark-plonk logic gates can perform bitwise AND and XOR operations.
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

File: plonk-book/src/chapter_3.md [hunk 10]
```diff
@@ -215,7 +213,7 @@ The result of the operation ends up in the 4th wire of the last subgate.
 The concept of lookup tables is to use precomputed tables in Plonk which allows circuits to be encoded more efficiently. These tables contain the most commonly used computation units which are very expensive to compute each time like XOR or AND operation between bit strings, or common hash functions such as AES-128 or SHA-256...etc. This removes the aspect of circuit complexity needed to compute the function, so the complexity of the SNARK depends only on the size of the function's domain.
 
 The Plonk circuit would then contain another type of gates (lookup gates) besides the original addition, multiplication and constant gates.
-In this PLONK lib, we use PlonKup, a ZK-SNARK that integrates plookup into PlonK in an efficient way. Before we explain how PlonKup works, we will introduce the original [plookup]((https://eprint.iacr.org/2020/315.pdf) ) definition.
+In ARK-PLONK, we use PlonKup, a ZK-SNARK that integrates plookup into PlonK in an efficient way. Before we explain how PlonKup works, we will introduce the original [plookup]((https://eprint.iacr.org/2020/315.pdf) ) definition.
```
Comment (CPerezz) [inline plonk-book/src/chapter_3.md]: ditto

**Review by CPerezz — CHANGES_REQUESTED** [review]
Benchmark section in `chapter3.md` has inconsistent processor descriptions.

I suggest to take them from the actual README where the exact model is stated.

File: plonk-book/src/SUMMARY.md (hunk 1 above)
Comment (mathcrypto) [reply plonk-book/src/SUMMARY.md]: Of course :)

**Comment by mathcrypto** [comment]
> Benchmark section in `chapter3.md` has inconsistent processor descriptions.
> 
> I suggest to take them from the actual README where the exact model is stated.

will leave them out of chapter 3 then

**Comment by mathcrypto** [comment]
> 

On top of Master?

**Comment by mathcrypto** [comment]
> @mathcrypto Can you rebase on top of `main` so that the history is not massive?? Otherways there are an insane amount of commits because you're re-using the same branch for different PR's.

On top of Master?

**Comment by CPerezz** [comment]
@mathcrypto yes sorry. `master` correct!!

**Review by LukePearson1 — COMMENTED** [review]
Will merge this, with a squash into 1 commit. 

The failing docs-link is noted in #96

**Pull request merged**